File: Homework2/src/thermostat/thermostatTest.java
package thermostat;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;


class thermostatTest {

    @org.junit.jupiter.api.BeforeEach
    void setUp() {
    }

    @org.junit.jupiter.api.AfterEach
    void tearDown() {
    }

    @Test
    public void testConstructor() {
        thermostat unit = new thermostat(36, tempType.C, fanSpeed.MEDIUM);
        assertTrue(unit.getTemp() == 36 && unit.getTempType() == tempType.C && unit.getFanSpeed() == fanSpeed.MEDIUM);
    }

    @Test
    public void testDefaultConstructor() {
        thermostat unit = new thermostat();
        assertTrue(unit.getTemp() == 75 && unit.getTempType() == tempType.F && unit.getFanSpeed() == fanSpeed.AUTO);
    }

    @Test
    public void testSetTemp() {
        thermostat unit = new thermostat();
        unit.setTemp(45);
        int expResult = 45;
        assertEquals(expResult, unit.getTemp());
    }

    @Test
    public void testSetTempInvalid() {
        thermostat unit = new thermostat();
        boolean ok = false;
        try {
            unit.setTemp(-1);
        } catch(IllegalArgumentException ex) {
            ok = true;
        }
        assertTrue(ok);
    }

    @Test
    public void testSetTempType() {
        thermostat unit = new thermostat();
        unit.setTempType(tempType.C);
        tempType expResult = tempType.C;
        assertEquals(unit.getTempType(), expResult);
    }

    @Test
    public void testSetFanSpeed() {
        thermostat unit = new thermostat();
        unit.setFanSpeed(fanSpeed.HIGH);
        fanSpeed expResult = fanSpeed.HIGH;
        assertEquals(unit.getFanSpeed(), expResult);
    }

}

File: Homework 1/thermostattest.py
import unittest
from thermostat import thermostat

class thermostattest(unittest.TestCase):

    def testinit(self):
        unit = thermostat()
        self.assertTrue(unit.getTemp() == 75 and unit.getTempType() == "F" and unit.getFanStrength() == thermostat.FAN_AUTO, "Initial values for unit are incorrect")

    def testSetTemp(self):
        unit = thermostat()
        unit.setTemp(50)
        expresult = 50
        self.assertEqual(unit.getTemp(), expresult, "temp should be 50")

    def testSetTempType(self):
        unit = thermostat()
        unit.setTempType("C")
        expresult = "C"
        self.assertEqual(unit.getTempType(), expresult, "tempType should be 'C'")

    def testSetFanStrength(self):
        unit = thermostat()
        unit.setFanStrength(thermostat.FAN_OFF)
        expresult = thermostat.FAN_OFF
        self.assertEqual(unit.getFanStrength(), expresult, "fanStrength should be FAN_OFF")

    def testSetTempInvalid(self):
        unit = thermostat()
        OK = False
        try:
            unit.setTemp("abcd")
        except ValueError:
            OK = True
        self.assertTrue(OK)

    def testSetTempTypeInvalid(self):
        unit = thermostat()
        OK = False
        try:
            unit.setTempType(75)
        except ValueError:
            OK = True
        self.assertTrue(OK)

    def testSetFanStrengthInvalid(self):
        unit = thermostat()
        OK = False
        try:
            unit.setFanStrength("FAN_OFF")
        except ValueError:
            OK = True
        self.assertTrue(OK)

if __name__ == '__main__':
    unittest.main()
